### PR TITLE
Mostgoals

### PR DESCRIPTION
**Pull request by MillsProvosty**

### CODE REVIEW

File: lib/team_statistics.rb [hunk 1]
```diff
@@ -51,4 +51,48 @@ def worst_season(team_id)
...
+
+    games_by_home_team_id = @games.find_all do |game|
+        game.home_team_id == team_id
+    end
+
+    min_away_goal = games_by_away_team_id.min_by do |game|
+      game.away_goals
+    end
+    min_away_goal = min_away_goal.away_goals
+
```
Comment (milevy1) [inline lib/team_statistics.rb:93]: Delete extra blank lines here

File: lib/team_statistics.rb [hunk 2]
```diff
@@ -51,4 +51,48 @@ def worst_season(team_id)
...
+    max_away_goal = games_by_away_team_id.max_by do |game|
+      game.away_goals
+    end
+    max_away_goal = max_away_goal.away_goals
+
+    max_home_goal = games_by_home_team_id.max_by do |game|
+      game.home_goals
+    end
+      max_home_goal = max_home_goal.home_goals
+
```
Comment (milevy1) [inline lib/team_statistics.rb:72]: Delete 1 extra blank line here

File: test/team_statistics_test.rb [hunk 3]
```diff
@@ -25,4 +25,14 @@ def test_worst_season
     assert_equal 20122013, @stat_tracker.worst_season(3)
     assert_equal 20152016, @stat_tracker.worst_season(30)
   end
+
+  def test_most_goals_scored
+
```
Comment (milevy1) [inline test/team_statistics_test.rb]: delete 1 extra blank line here

File: test/team_statistics_test.rb [hunk 4]
```diff
@@ -25,4 +25,14 @@ def test_worst_season
...
+    assert_equal 5, @stat_tracker.most_goals_scored(6)
+  end
+
+  def test_fewest_goals_scored
+
```
Comment (milevy1) [inline test/team_statistics_test.rb]: delete this extra line too

File: test/team_statistics_test.rb (hunk 3 above)
Comment (MillsProvosty) [reply test/team_statistics_test.rb]: Done and done.

File: test/team_statistics_test.rb (hunk 4 above)
Comment (MillsProvosty) [reply test/team_statistics_test.rb]: Got it.

**Pull request merged**